tools/tidy/tests/CMakeLists.txt

```diff
@@ -5,9 +5,26 @@
 
 add_executable(
   tidy_unittests
-  ../../../tests/unittests/main.cpp ../../../tests/unittests/Test.cpp
-  ../synthesis/OnlyAssignedOnReset.cpp OnlyAssignedOnResetTest.cpp
-  ../synthesis/RegisterHasNoReset.cpp RegisterHasNoResetTest.cpp)
+  ../../../tests/unittests/main.cpp
+  ../../../tests/unittests/Test.cpp
+  ../src/ASTHelperVisitors.cpp
```

Comment:
It's probably worth compiling these into an object library at least, so that you aren't compiling each file twice in the build (once for the tool and once for the tests).

Reply:
If you are referring to the checks `cpp` files, I'm not sure I can do that since then the static variable that registers the checks gets optimized out by the linker. I'm talking about the `REGISTER` macro I use to register the different checks into the factory

Reply:
I think if you use an OBJECT library that shouldn't happen? It's just a collection of compiled object files that get pulled into each build.